src/components/LogInForm/form.jsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import "./logInForm.css";
+
+export default class Form extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      email: "",
+      nameError: "",
+      emailError: "",
+      send: false,
+      errors: []
+    };
+  }
+
+  handleChange(e) {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    if (this.validate()) {
+      this.setState({ send: true });
+    }
+  }
+
+  validate() {
+    let nameError = "";
+    let emailError = "";
+    let errors = [];
+
+    if (this.state.name.length < 2 && this.state.name.indexOf(" ") === -1) {
+      nameError = "Your login is not correct!";
+      errors.push(nameError);
+    }
```

Comment:
This way validation most definitely didn't come up as it should. It fails only if the name is shorter than 2 characters OR if it contains no spaces. A single space, like this: " " is a valid name according to this validation.